stix2/v21/observables.py

```diff
@@ -986,6 +999,26 @@ class X509Certificate(_Observable):
     ])
     _id_contributing_properties = ["hashes", "serial_number"]
 
+    def _check_object_constraints(self):
+        super(X509Certificate, self)._check_object_constraints()
+
+        if (
+                self.get('is_self_signed') is None and
+                self.get('hashes') is None and
+                self.get('version') is None and
+                self.get('serial_number') is None and
+                self.get('signature_algorithm') is None and
+                self.get('issuer') is None and
+                self.get('validity_not_before') is None and
+                self.get('validity_not_after') is None and
+                self.get('subject') is None and
+                self.get('subject_public_key_algorithm') is None and
+                self.get('subject_public_key_modulus') is None and
+                self.get('subject_public_key_exponent') is None and
+                self.get('x509_v3_extensions') is None
+        ):
```

Comment:
I think we could use `self._check_at_least_one_property()` here, similar to https://github.com/oasis-open/cti-python-stix2/blob/d4c01157352552e2ec05bcf7b0f79ef96e34022d/stix2/v20/common.py#L49